src/functions/lint_package/dotfiles.sh

```diff
@@ -28,11 +28,10 @@ source "$LIBRARY/util/message.sh"
 lint_package_functions+=('check_dotfiles')
 
 check_dotfiles() {
-	local ret=0
-	for f in "$pkgdir"/.*; do
-		[[ ${f##*/} == . || ${f##*/} == .. ]] && continue
-		error "$(gettext "Dotfile found in package root '%s'")" "$f"
-		ret=1
-	done
-	return $ret
+    local ret=0
+    for f in $(ls -Ad "$pkgdir"/.*); do
```

Comment:
We should be able to just avoid `ls` completely by using something like this. It does add another variable into the function, but it stops all this globbing nonsense that's causing issues:

```sh
local dotfiles
mapfile -t dotfiles < <(find -mindepth 1 -maxdepth 1 | sed 's|^\./||' | grep '^\.')
```

Reply:
What if I replace line 32 with
```bash
for f in $(find "$pkgdir" -mindepth 1 -maxdepth 1 -name '.*'); do
```
It also eliminate the globing issue and also doesn't introduce any new variable. I think it also better to keep the same error message wherein the full path of the dotfile is displayed.

Reply:
If there's a dotfile named something like `.one two`, then that gets printed as two separate files with that approach. It's not the end of the world if that were to happen, but I'd prefer to stick with `mapfile` just so errors all line up properly.